### PR TITLE
Align AWQ with reference (mean stat, weight-aware scale, clipping)

### PR DESCRIPTION
**Pull request by JyotinderSingh**
## Description

The Keras AWQ implementation diverged from the reference algorithms (llm-awq / AutoAWQ) in three ways that hurt quantized accuracy:
1. The activation statistic was a running MAX of |x|, whereas reference AWQ uses the per-channel MEAN of |x|. Max over-weights a single outlier token and destabilizes the saliency estimate.
2. The scale search used `scales = x_max**ratio` only, dropping the weight-magnitude term. The reference formula balances activation and weight magnitude: `scales = x_stat**ratio / w_stat**(1-ratio)`.
3. There was no weight-clipping step. AutoAWQ additionally grid-searches a per-group clip on the weight max to minimize reconstruction error.

### Changes

`keras/src/quantizers/awq.py`:
- `update_activation_magnitudes` now keeps a running per-channel MEAN of |x| via a numerically stable, batch-count-weighted incremental update (`mean += (batch_mean - mean) * n / (count + n)`). It also stashes a bounded raw-activation sample (`MAX_CLIP_SAMPLE_ROWS=512`) for clipping.
- `awq_search_optimal_scales` now uses the reference scale formula with the new `_get_weight_scale()` term (|W| normalized per group, averaged over the output axis, as in llm-awq get_weight_scale). The existing `1e-4` clamp and `sqrt(max*min)` normalization and the ratio-in-[0,1] grid are kept; the activation-weighted MSE selection loss is unchanged.
- New `awq_search_best_clip()` implements AutoAWQ's best-clip: for each output channel and group it grid-searches ~20 shrink factors on the per-group max and keeps the value minimizing reconstruction MSE of the layer output against the stashed sample (reconstruction uses the scaled input x/awq_scales to match inference). Output channels are batched to bound memory. Gated by AWQConfig.apply_clip and applied after the scale search, before final quantization.
- `_fake_quantize_weights()` factors the shared quantize/dequantize path so the scale search, clip search, and final quantization all evaluate the exact same 4-bit kernel.

`keras/src/quantizers/awq_config.py`:
- New serialized knob `apply_clip (default True)`, round-tripped in `get_config`/`from_config`.

### Evidence

[[Script](https://gist.github.com/JyotinderSingh/6f25918c0e683a174db33bd8f07f92b0)]

Both old and new reuse the identical 4-bit quantizer kernel; only the AWQ scale/statistic/clipping logic differs.

1. Per-layer activation-weighted reconstruction error mean((W - W_recon)^2 * x_mean^2) over 18 randomized layers with injected outlier channels (out/in/group in {64..256}/{128..512}/ {-1,32,64,128}):
```
      NEW <= OLD on 18/18 layers.
      Mean error OLD=1.102e-01, NEW=9.701e-02 (ratio 0.880, ~12% lower).
      Per-layer NEW/OLD ratios span 0.688..0.985.
```

2. Tiny end-to-end transformer classifier (awq_test.py pattern, scaled to embed=128, heads=8, ff=384, 2 blocks, gs=128, with channel outliers baked into the block input), averaged over 5 model seeds, KL / top-1 agreement vs the float model (lower KL / higher top-1 better):
```
      mean KL      OLD=0.00278  NEW=0.00258  (0.930x, ~7% lower)
      mean top-1   OLD=0.800    NEW=0.850
      NEW beats OLD on both averaged metrics.
```

### Non-goals / Caveats
- Non-goal: folding awq_scales into stored weights / preceding ops using fused kernels is deferred.
- The clip search minimizes per-(output-channel, group) partial-output MSE independently (as in AutoAWQ), not the full summed-output MSE.
- Clipping falls back to a single per-output-channel group when the group size does not evenly divide in_features.

## Contributor Agreement

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

File: keras/src/quantizers/awq_test.py [hunk 1]
```diff
@@ -77,6 +79,91 @@ def test_scale_search_with_zero_activations(self):
...
+        """Clipping must not increase the per-group output recon error.
+
+        The clip search selects, per (output channel, group), the shrink factor
+        minimizing the reconstruction error of that group's partial output
+        against the (unclipped) scaled weights. Because the search grid always
+        includes the no-shrink candidate, this error can only decrease. This
+        test measures that exact objective with vs without clipping on weights
+        with injected outliers.
+        """
+        from keras.src.quantizers.quantizers import dequantize_with_sz_map
```
Comment (jeffcarp) [inline keras/src/quantizers/awq_test.py:124]: This can probably just be up top?

File: keras/src/quantizers/awq_test.py [hunk 2]
```diff
@@ -291,23 +383,68 @@ def test_awq_activation_accumulation(self):
...
+        expected_mean = ops.mean(combined, axis=0)
+        self.assertEqual(awq_obj.num_samples, 40)
         self.assertAllClose(
-            awq_obj.activation_magnitudes, expected_max, atol=1e-6
+            awq_obj.activation_magnitudes, expected_mean, atol=1e-6
+        )
+
+    def test_awq_clip_sample_capture(self):
+        """Test that a bounded activation sample is stashed for clipping."""
+        from keras.src.quantizers.awq import MAX_CLIP_SAMPLE_ROWS
```
Comment (jeffcarp) [inline keras/src/quantizers/awq_test.py:404]: Same

**Review by jeffcarp — APPROVED** [review]

File: keras/src/quantizers/awq_test.py (hunk 1 above)
Comment (JyotinderSingh) [reply keras/src/quantizers/awq_test.py:124]: This causes a circular import, which is why it is imported locally. We use this pattern in a few other quantization sites too.

File: keras/src/quantizers/awq_test.py (hunk 2 above)
Comment (JyotinderSingh) [reply keras/src/quantizers/awq_test.py:404]: This causes a circular import, which is why it is imported locally. We use this pattern in a few other quantization sites too.
